ddtrace/context.py

```diff
@@ -45,67 +19,64 @@ def __init__(
     ):
         # type: (...) -> None
         """
-        Initialize a new thread-safe ``Context``.
-
         :param int trace_id: trace_id of parent span
         :param int span_id: span_id of parent span
         """
-        self._trace = []  # type: List[Span]
-        self._finished_spans = 0
-        self._current_span = None  # type: Optional[Span]
-        self._lock = threading.Lock()
```

Comment:
I don't think the lock can be removed here as a `Context` can still be shared across threads